backend/src/controllers/MyUserController.ts

```diff
@@ -32,23 +37,26 @@ const updateCurrentUser = async (req: Request, res: Response) => {
     user.city = city;
 
     await user.save();
+    logger.info(`User updated with ID ${req.userId}`);
     res.send(user);
   } catch (error) {
-    console.log(error);
-    res.status(500).json({ message: 'Error updating user' });
+    next(error);
   }
 };
 
-const getCurrentUser = async (req: Request, res: Response) => {
+const getCurrentUser = async (req: Request, res: Response, next: NextFunction) => {
   try {
     const currentUser = await User.findById({ _id: req.userId });
+
     if (!currentUser) {
+      logger.warn(`User fetch failed: User not found with ID ${req.userId}`);
       return res.status(404).send('User not found');
     }
+
+    logger.info(`Fetched current user with ID ${req.userId}`);
     res.json(currentUser);
   } catch (error) {
-    console.log(error);
-    res.status(500).send('Error fetching user');
+    next(error);
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider more specific error handling</b></div>
<div id="fix">

Consider adding more specific error handling. Instead of passing the error directly to `next()`, you might want to log the error and send a more user-friendly response. A similar issue was also found in backend/src/controllers/MyUserController.ts (line 43).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -58,2 +58,4 @@
    } catch (error) {
 -    next(error);
 +    logger.error(`Error fetching current user: ${error.message}`);
 +    res.status(500).json({ message: 'An error occurred while fetching the user' });
 +    next(error);
    }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/Food-Ordering-App/pull/47#issuecomment-2439370000>#eef1bf</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged